File: LMIS/LMIS.Web/Modules/BookManage/Bookshelfs/BookshelfsColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace LMIS.BookManage.Columns;

[ColumnsScript("BookManage.Bookshelfs")]
[BasedOnRow(typeof(BookshelfsRow), CheckNames = true)]
public class BookshelfsColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public long BookshelfId { get; set; }
    [EditLink]
    public string Location { get; set; }
    public string CategoryName { get; set; }
    public string BookshelfImage { get; set; }
    public long BookCount { get; set; }
    public long BookCapacity { get; set; }
    public DateTime CreateTime { get; set; }
    public DateTime UpdateTime { get; set; }
}

File: LMIS/LMIS.Web/Modules/ServerTypes/BookManage/BookshelfsColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { BookshelfsRow } from "./BookshelfsRow";

export interface BookshelfsColumns {
    BookshelfId: Column<BookshelfsRow>;
    Location: Column<BookshelfsRow>;
    CategoryName: Column<BookshelfsRow>;
    BookshelfImage: Column<BookshelfsRow>;
    BookCount: Column<BookshelfsRow>;
    BookCapacity: Column<BookshelfsRow>;
    CreateTime: Column<BookshelfsRow>;
    UpdateTime: Column<BookshelfsRow>;
}

export class BookshelfsColumns extends ColumnsBase<BookshelfsRow> {
    static readonly columnsKey = 'BookManage.Bookshelfs';
    static readonly Fields = fieldsProxy<BookshelfsColumns>();
}